app/src/main/java/com/x8bit/bitwarden/ui/platform/components/toggle/BitwardenUnlockWithBiometricsSwitch.kt

```diff
@@ -17,23 +24,46 @@ import com.x8bit.bitwarden.R
  */
 @Composable
 fun BitwardenUnlockWithBiometricsSwitch(
-    isBiometricsSupported: Boolean,
+    biometricSupportStatus: BiometricSupportStatus,
     isChecked: Boolean,
     onDisableBiometrics: () -> Unit,
     onEnableBiometrics: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
-    if (!isBiometricsSupported) return
-    BitwardenWideSwitch(
-        modifier = modifier,
-        label = stringResource(id = R.string.unlock_with, stringResource(id = R.string.biometrics)),
-        isChecked = isChecked,
-        onCheckedChange = { toggled ->
-            if (toggled) {
-                onEnableBiometrics()
-            } else {
-                onDisableBiometrics()
+    if (biometricSupportStatus == BiometricSupportStatus.NOT_SUPPORTED) return
+    Column(modifier = modifier) {
+        BitwardenWideSwitch(
+            label = stringResource(
+                id = R.string.unlock_with,
+                stringResource(id = R.string.biometrics),
+            ),
+            isChecked = isChecked,
+            onCheckedChange = { toggled ->
+                if (toggled) {
+                    onEnableBiometrics()
+                } else {
+                    onDisableBiometrics()
+                }
+            },
+            enabled = biometricSupportStatus == BiometricSupportStatus.CLASS_3_SUPPORTED,
+        )
+        val biometricsDescription: String = when (biometricSupportStatus) {
+            BiometricSupportStatus.CLASS_3_SUPPORTED -> {
+                stringResource(R.string.class_3_biometrics_description)
             }
-        },
-    )
+            BiometricSupportStatus.CLASS_2_SUPPORTED -> {
+                stringResource(R.string.class_2_biometrics_description)
+            }
+
+            BiometricSupportStatus.NOT_SUPPORTED -> error(
+                "Should not be called when BiometricSupportStatus is NOT_SUPPORTED",
+            )
+        }
+        Text(
```

Comment:
Instead of this custom `Text` can we just pass in the string as the `description` of the `BitwardenWideSwitch`

Reply:
way easier lol, thank you!